internal/pkg/service/stream/storage/level/staging/file.go

```diff
@@ -1,28 +1,34 @@
 package staging
 
 import (
-	"github.com/keboola/go-client/pkg/keboola"
+	"time"
 
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/compression"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local"
 )
 
+const (
+	DefaultFileExpiration = 24 * time.Hour
+)
+
 type File struct {
 	// IsEmpty is set if the import was skipped because there is no slice in the file.
 	IsEmpty bool `json:"isEmpty,omitempty"`
+	// Provider of the staging file.
+	Provider FileProvider `json:"provider" validate:"required"`
 	// Compression configuration.
-	Compression compression.Config `json:"compression"  `
-	// UploadCredentials to the staging storage.
-	UploadCredentials           *keboola.FileUploadCredentials `json:"credentials" validate:"required"`
-	UploadCredentialsExpiration utctime.UTCTime                `json:"credentialsExpiration" validate:"required"`
+	Compression compression.Config `json:"compression" validate:"required"`
+	// Expiration determines how long it is possible to write to the staging file, e.g. due to expiring credentials.
+	Expiration utctime.UTCTime `json:"expiration" validate:"required"`
```

Comment:
File/Slice entity is now more general, I removed Keboola specific fields, and added `Provider` field.

The upload credentials are managed by the `keboolasink.Bridge` and have own etcd prefix.